xonsh/prompt/vc.py

```diff
@@ -20,9 +20,13 @@
     "RE_REMOVE_ANSI",
 )
 
+# when running git status commands we do not want to acquire locks running command like git status
+_GIT_OPTIONAL_LOCKS = {"GIT_OPTIONAL_LOCKS": "0"}
+
 
 def _get_git_branch(q):
     denv = builtins.__xonsh__.env.detype()
+    denv.update(_GIT_OPTIONAL_LOCKS)
```

Comment:
You shouldn't modify the detyped env directly. This object can be shared accidientally.  It would be better to wrap the `_get_git_branch()` call in an `with builtins.__xonsh__.env.swap(GIT_OPTIONAL_LOCKS=0):` context